src/lib/actions/redirectWithIdentity/createRedirectWithIdentity.js

```diff
@@ -10,11 +10,13 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-module.exports = ({ instanceManager, document, logger }) => (
+const createConfigOverrides = require("../../utils/createConfigOverrides");
```

Comment:
Any time you require something in a file, it can make unit testing harder. In this case, in order to test that the correct overrides are used, you end up testing the logic in createConfigOverrides too. The tests stop being unit tests. For example, if there is a bug in createConfigOverrides, then a bunch of the tests are going to fail. There is a balance though because javascript's built in functions are quite limited. So sometimes it is okay to require a utility.

So what is properly a utility? Is it anything that is in the "utils" folder? Maybe. Certainly it needs to be a pure function, and createConfigOverrides is a pure function. I think it's important also that a util is something that has been thoroughly tested. I think what fails here though is that a utility should be something that is generally applicable, and not something that implements specific business logic. (i.e. a map function, a find, or assign).

Another way I like to think about it is that if we are following dependency injection principles, we should be able to separate the code into units that create new instances of objects and units that don't. The standard we've come to within the Alloy repos is that "index.js" files can create object/functions, and import/require other files, while the rest of the code tries to stay within it's unit.

So here is a different approach. Change the signature of createConfigOverrides to this:
```
const createGetConfigOverrides = environmentName => settings => {
```
Then in your index.js files do this:
```
const createRedirectWithIdentity = require("./createRedirectWithIdentity");
const instanceManager = require("../../instanceManager/index");
const createGetConfigOverrides = require("../../utils/createGetConfigOverrides");

const getConfigOverrides = createGetConfigOverrides(turbine.environment.stage);
const document = window.document;
module.exports = createRedirectWithIdentity({
   instanceManager,
   document,
   logger: turbine.logger,
   getConfigOverrides
});
```
So now you are passing a function to extract config overrides out of a settings object instead of having to pass the whole turbine object and having to require createConfigOverrides. Now when you run unit tests on createRedirectWithIdentity, you only have to verify that the getConfigOverrides function gets all the settings, and that whatever is returned from getConfigOverrides is sent to redirectWithIdentity.